mmedit/models/components/discriminators/smpatch_disc.py

```diff
@@ -0,0 +1,128 @@
+        num_conv (int): Number of stacked intermediate convs (excluding input
+            and output conv). Default: 3.
+        norm_cfg (dict): Config dict to build norm layer.
+        init_cfg (dict): Config dict for initialization.
+            `type`: The name of our initialization method. Default: 'normal'.
+            `gain`: Scaling factor for normal, xavier and orthogonal.
+            Default: 0.02.
+        with_spectral_norm (bool): Whether use spectral norm after the conv
+            layers. Default to False.
+    """
+
+    def __init__(self,
+                 in_channels,
+                 base_channels=64,
+                 num_conv=3,
+                 norm_cfg=None,
+                 init_cfg=dict(type='normal', gain=0.02),
+                 with_spectral_norm=False):
+        super().__init__()
+
+        kernel_size = 4
+        padding = 1
+
+        # input layer
+        sequence = [
+            ConvModule(
+                in_channels=in_channels,
+                out_channels=base_channels,
+                kernel_size=kernel_size,
+                stride=2,
+                padding=padding,
+                bias=False,
+                norm_cfg=norm_cfg,
+                act_cfg=dict(type='LeakyReLU', negative_slope=0.2),
+                with_spectral_norm=with_spectral_norm)
+        ]
+
+        # stacked intermediate layers,
+        # gradually increasing the number of filters
+        multiple_now = 1
+        multiple_prev = 1
+        for n in range(1, num_conv):
+            multiple_prev = multiple_now
+            multiple_now = min(2**n, 8)
+            sequence += [
+                ConvModule(
+                    in_channels=base_channels * multiple_prev,
+                    out_channels=base_channels * multiple_now,
+                    kernel_size=kernel_size,
+                    stride=2,
+                    padding=padding,
+                    bias=False,
+                    norm_cfg=norm_cfg,
+                    act_cfg=dict(type='LeakyReLU', negative_slope=0.2),
+                    with_spectral_norm=with_spectral_norm)
+            ]
+        multiple_prev = multiple_now
+        multiple_now = min(2**num_conv, 8)
+        sequence += [
+            ConvModule(
+                in_channels=base_channels * multiple_prev,
+                out_channels=base_channels * multiple_now,
+                kernel_size=kernel_size,
+                stride=1,
+                padding=padding,
+                bias=False,
+                norm_cfg=norm_cfg,
+                act_cfg=dict(type='LeakyReLU', negative_slope=0.2),
+                with_spectral_norm=with_spectral_norm)
+        ]
+
+        # output one-channel prediction map
+        sequence += [
+            build_conv_layer(
+                dict(type='Conv2d'),
+                base_channels * multiple_now,
+                1,
+                kernel_size=kernel_size,
+                stride=1,
+                padding=padding)
```

Comment:
So here is a conventional `nn.Conv2d`?

Reply:
yes

Reply:
Then I think we can directly use `nn.Conv2d`.